Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -1381,6 +1381,36 @@ There are several ways to perform dynamic analysis:
 1. For the actual persistence: Use the techniques described in the data storage chapter.
 2. For reflection-based approaches: Use Xposed to hook into the deserialization methods or add unprocessable information to the serialized objects to see how they are handled (e.g., whether the application crashes or extra information can be extracted by enriching the objects).
 
+### Testing for Tapjacking Vulnerability (MSTG-PLATFORM-9)
+
+#### Overview
+
+Tapjacking is a vulnerability in which the malicious application abuses the screen overlay feature of Android to capture user taps and keystrokes. This is possible because an active screen overlay can listen for taps and intercept any information being passed in the underlying activity.
+
+Over the years many known malwares like MazorBot, BankBot, MysteryBot have been abusing screen overlay feature to target business critical applications, namely in the banking sector. The [blog](https://www.infosecurity-magazine.com/opinions/overlay-attacks-safeguard-mobile/ "blog") discusses more about these type of malwares.
+
+#### Static Analysis
+
+Check the source code for implementation of any of the below techniques:
+
+- Override [`onFilterTouchEventForSecurity`](https://developer.android.com/reference/android/view/View#onFilterTouchEventForSecurity%28android.view.MotionEvent%29 "onFilterTouchEventForSecurity")
+- Set [`android:filterTouchesWhenObscured`](https://developer.android.com/reference/android/view/View.html#setFilterTouchesWhenObscured%28boolean%29 "android:filterTouchesWhenObscured") to true
+- [FLAG_WINDOW_IS_OBSCURED](https://developer.android.com/reference/android/view/MotionEvent.html#FLAG_WINDOW_IS_OBSCURED "FLAG_WINDOW_IS_OBSCURED")
+
+When either of the first two controls are implemented, the android framework will discard touches or taps that are received from any other visible window. This will safeguard users against this severe vulnerability.
```

Comment:
```suggestion
When either of the first two techniques are implemented, the Android framework will discard touches or taps that are received from any other visible window. This will safeguard users against this severe vulnerability.
```